rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry.scala

```diff
@@ -880,6 +885,35 @@ class RegistryImpl[F[_]](
       case _ => F.unit
     }
 
+  private def handleRhoIdLookup(
+      id: String,
+      sequenceNumber: Int,
+      ret: Par,
+      rand: Blake2b512Random,
+      localFail: () => F[Unit],
+      uri: String
+  ) =
+    if (id.size != 54) {
+      localFail()
+    } else {
+      // Could fail
+      // 256 bits plus 14 bit crc-14
+      val bytes: Array[Byte] = ZBase32.decode(id, 270)
```

Comment:
decode throws IllegalArgument exception. You should handle that exception locally and  return an Option